src/linear_algebra/sesquilinear_form.lean

```diff
@@ -363,6 +363,162 @@ lemma is_compl_span_singleton_orthogonal {B : V →ₗ[K] V →ₗ[K] K}
 
 end orthogonal
 
+/-! ### Adjoint pairs -/
+
+section adjoint_pair
+
+section add_comm_monoid
+
+variables [comm_semiring R]
+variables [add_comm_monoid M] [module R M]
+variables [add_comm_monoid M₁] [module R M₁]
+variables [add_comm_monoid M₂] [module R M₂]
+variables (B F : M →ₗ[R] M →ₗ[R] R) (B' : M₁ →ₗ[R] M₁ →ₗ[R] R) (B'' : M₂ →ₗ[R] M₂ →ₗ[R] R)
+variables (f f' : M →ₗ[R] M₁) (g g' : M₁ →ₗ[R] M)
+
+/-- Given a pair of modules equipped with bilinear forms, this is the condition for a pair of
+maps between them to be mutually adjoint. -/
+def is_adjoint_pair := ∀ ⦃x y⦄, B' (f x) y = B x (g y)
```

Comment:
Are the semi-implicit args useful here? Why not
```suggestion
def is_adjoint_pair := ∀ x y, B' (f x) y = B x (g y)
```
Which means you can write `h x y`

Reply:
Having semi-implicit arguments avoids these awkward `h _ _ hx hy` proof terms, because you can just write `h hx hy`.

Reply:
What are `hx` and `hy` here? I only see two binders